src/main/java/mgtow/storage/Storage.java

```diff
@@ -0,0 +1,58 @@
+package mgtow.storage;
+
+import mgtow.task.Task;
+import mgtow.util.InvalidTaskException;
+import mgtow.util.Parser;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        createDirectoryIfNotExists();
+    }
+
+    private void createDirectoryIfNotExists() {
+        File directory = new File(filePath).getParentFile();
+        if (!directory.exists()) {
+            boolean created = directory.mkdirs();
+            if (created) {
+                System.out.println("\tCreated directory: " + directory.getAbsolutePath());
+            } else {
+                System.out.println("\tFailed to create directory: " + directory.getAbsolutePath());
+            }
+        }
+    }
+
+    public ArrayList<Task> load() throws InvalidTaskException {
```

Comment:
Could we consider changing the name to be more specific? Like "loadTasks()" or "loadTasksFromStorage()"?